python/src/semantic_retrieval/utils/callbacks.py

```diff
@@ -77,6 +77,15 @@ def default(o: Any):
     return json.dumps(obj, default=default, **kwargs)
 
 
+def _maybe_redacted(key: str, value: Any) -> Any:
+    if isinstance(value, str) and any(
+        k in key.lower() for k in ["api_key", "apikey", "token", "secret"]
```

Comment:
Could we just regex on key, token, secret in case it's some thing like `API_TOKEN` for example?

Reply:
Agreed, but I don't think we can construct a perfect regexp for this. At least this is much better than no filtering

Reply:
FWIW, "API_TOKEN" will match already because "token" is in "API_TOKEN".lower().